components/eventing-controller/testing/test_helpers.go

```diff
@@ -65,6 +66,18 @@ const (
 
 type APIRuleOption func(r *apigatewayv1alpha1.APIRule)
 
+func GetFreePort() (port int, err error) {
+	var a *net.TCPAddr
+	if a, err = net.ResolveTCPAddr("tcp", "localhost:0"); err == nil {
+		var l *net.TCPListener
+		if l, err = net.ListenTCP("tcp", a); err == nil {
+			defer l.Close()
```

Comment:
I actually started o use the `GetFreePort` func :D. I also added err handling via gomega
```
	natsPort, err := reconcilertesting.GetFreePort()
	g.Expect(err).ToNot(gomega.HaveOccurred())
	ens := setupTestEnsemble(ctx, reconcilertesting.EventTypePrefix, g, natsPort)
```

since the  `metricsPort` is not needed anywhere else it now gets created where metrics gets started.